internal/resource/config/oauth/oauth_auth_server_settings_resource.go

```diff
@@ -0,0 +1,1167 @@
+	var plan oauthAuthServerSettingsResourceModel
+
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	createOauthAuthServerSettings := client.NewAuthorizationServerSettings(plan.DefaultScopeDescription.ValueString(), plan.AuthorizationCodeTimeout.ValueInt64(), plan.AuthorizationCodeEntropy.ValueInt64(), plan.RefreshTokenLength.ValueInt64(), plan.RefreshRollingInterval.ValueInt64(), plan.RegisteredAuthorizationPath.ValueString(), plan.PendingAuthorizationTimeout.ValueInt64(), plan.DevicePollingInterval.ValueInt64(), plan.BypassActivationCodeConfirmation.ValueBool())
+	err := addOptionalOauthAuthServerSettingsFields(ctx, createOauthAuthServerSettings, plan)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to add optional properties to add request for OauthAuthServerSettings", err.Error())
+		return
+	}
+	requestJson, err := createOauthAuthServerSettings.MarshalJSON()
+	if err == nil {
+		tflog.Debug(ctx, "Add request: "+string(requestJson))
+	}
+
+	apiCreateOauthAuthServerSettings := r.apiClient.OauthAuthServerSettingsApi.UpdateAuthorizationServerSettings(config.ProviderBasicAuthContext(ctx, r.providerConfig))
+	apiCreateOauthAuthServerSettings = apiCreateOauthAuthServerSettings.Body(*createOauthAuthServerSettings)
+	oauthAuthServerSettingsResponse, httpResp, err := r.apiClient.OauthAuthServerSettingsApi.UpdateAuthorizationServerSettingsExecute(apiCreateOauthAuthServerSettings)
+	if err != nil {
+		config.ReportHttpError(ctx, &resp.Diagnostics, "An error occurred while creating the OauthAuthServerSettings", err, httpResp)
+		return
+	}
+	responseJson, err := oauthAuthServerSettingsResponse.MarshalJSON()
+	if err == nil {
+		tflog.Debug(ctx, "Add response: "+string(responseJson))
+	}
+
+	// Read the response into the state
+	var state oauthAuthServerSettingsResourceModel
+
+	readOauthAuthServerSettingsResponse(ctx, oauthAuthServerSettingsResponse, &state)
+	diags = resp.State.Set(ctx, state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (r *oauthAuthServerSettingsResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	readOauthAuthServerSettings(ctx, req, resp, r.apiClient, r.providerConfig)
+}
+
+func readOauthAuthServerSettings(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse, apiClient *client.APIClient, providerConfig internaltypes.ProviderConfiguration) {
+	var state oauthAuthServerSettingsResourceModel
+
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	apiReadOauthAuthServerSettings, httpResp, err := apiClient.OauthAuthServerSettingsApi.GetAuthorizationServerSettings(config.ProviderBasicAuthContext(ctx, providerConfig)).Execute()
+
+	if err != nil {
+		config.ReportHttpError(ctx, &resp.Diagnostics, "An error occurred while looking for a OauthAuthServerSettings", err, httpResp)
```

Comment:
As I learned recently, we shouldn't report an error on read for a 404 https://developer.hashicorp.com/terraform/plugin/framework/resources/read#recommendations

The same applies to delete  https://developer.hashicorp.com/terraform/plugin/framework/resources/delete#recommendations

Reply:
404 check added in the latest commit